frontend/src/ArticleEdit/ArticleEdit.js

```diff
@@ -2,81 +2,113 @@ import React from 'react';
 import marked from 'marked';
 import PropTypes from 'prop-types';
 
+import actions from './actions';
+
 import Input from './Input';
 import Tab from './Tab';
-import { Button, Editor, Heading1, Preview, Textarea, Wrapper } from './styled';
+import {
+  Button,
+  Editor,
+  Heading1,
+  Preview,
+  Textarea,
+  Wrapper,
+  Message,
+} from './styled';
 
 const propTypes = {
-  data: PropTypes.shape({
-    topic: PropTypes.string,
-    content: PropTypes.string,
-    sub_topic: PropTypes.array,
-    title: PropTypes.string,
-  }),
+  id: PropTypes.string,
   empty: PropTypes.bool,
 };
 
 const defaultProps = {
-  data: { topic: '', content: '', sub_topic: [], title: '' },
+  id: '0',
   empty: false,
 };
 
 class ArticleEdit extends React.Component {
   state = {
     edit: true,
-    content: this.props.empty ? '' : this.props.data.content,
+    article: {},
+    message: { show: false, error: false, text: '' },
+  };
+
+  componentDidMount = () => {
+    if (this.props.id && this.props.id !== '0') {
```

Comment:
If the default of `id` is '0', `this.props.id` will always be `true`.
By setting `defaultProps` of `id` to an empty string, we can get rid of the `this.props.id !== '0'`

Reply:
if we default to empty string, this.props.id will still always be true?
Can't set it to false or null because we get invalid prop-type, hence the default we can check.
Correct me if I'm wrong, I'm never confident about these sorts of things.
We could generalise the prop types I suppose, would that be preferable?

Reply:
Boolean of an empty string coerces to `false`.
Btw, PropTypes can be set to differen types with `oneOfType` which takes an array of types :wink:

Reply:
was sure I checked the boolean cast earlier in console, but it's certainly falsey now :blush: ... Will fix.